library/javatests/net/openid/appauth/IdTokenTest.java

```diff
@@ -0,0 +1,92 @@
+package net.openid.appauth;
+
+import android.support.annotation.Nullable;
+import android.util.Base64;
+
+import net.openid.appauth.IdToken.IdTokenException;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk=16)
+public class IdTokenTest {
+
+    static final String TEST_ID_TOKEN = "eyJhbGciOiJSUzI1NiIsImtpZCI6IjFlOWdkazcifQ.ewogIml"
+    + "zcyI6ICJodHRwOi8vc2VydmVyLmV4YW1wbGUuY29tIiwKICJzdWIiOiAiMjQ4Mjg5NzYxMDAxIiwKICJhdWQiOiAiczZ"
+    + "CaGRSa3F0MyIsCiAibm9uY2UiOiAibi0wUzZfV3pBMk1qIiwKICJleHAiOiAxMzExMjgxOTcwLAogImlhdCI6IDEzMTE"
+    + "yODA5NzAKfQ.ggW8hZ1EuVLuxNuuIJKX_V8a_OMXzR0EHR9R6jgdqrOOF4daGU96Sr_P6qJp6IcmD3HP99Obi1PRs-cw"
+    + "h3LO-p146waJ8IhehcwL7F09JdijmBqkvPeB2T9CJ NqeGpe-gccMg4vfKjkM8FcGvnzZUN4_KSP0aAp1tOJ1zZwgjxq"
+    + "GByKHiOtX7TpdQyHE5lcMiKPXfEIQILVq0pc_E2DzL7emopWoaoZTF_m0_N0YzFC6g6EJbOEoRoSK5hoDalrcvRYLSrQ"
+    + "AZZKflyuVCyixEoV9GfNQC3_osjzw2PAithfubEEBLuVVk4XUVrWOLrLl0nx7RkKU8NXNHq-rvKMzqg";
+    static final String TEST_SUBJECT = "SUBJ3CT";
+
+
+    @Test
+    public void testFrom() throws Exception {
+        IdToken idToken = IdToken.from(TEST_ID_TOKEN);
```

Comment:
Looks like you're missing some assertions on the expected content of the ID token here.